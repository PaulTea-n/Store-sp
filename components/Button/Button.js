import React from "react";
import styles from "./Button.module.scss";
import PropTypes from "prop-types";

const Button = (props) => {
  const { text, type, handleClick, backgroundColor } = props;
  return (
    <button
      className={styles.btn}
      onClick={handleClick}
      type={type}
      style={{ backgroundColor: backgroundColor }}
    >
      {text}
    </button>
  );
};

Button.propTypes = {
  text: PropTypes.string,
  handleClick: PropTypes.func,
  backgroundColor: PropTypes.string,
  type: PropTypes.oneOf(["button", "submit", "reset"]),
};

Button.defaultProps = {
  text: "Add to cart",
  backgroundColor: "red",
  type: "button",
  handleClick: () => {},
};

export default Button;
