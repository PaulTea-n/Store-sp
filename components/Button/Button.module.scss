.btn {
    box-sizing: border-box;
    padding: 12px;
    font-size: 16px;
    border-radius: 10px;
    margin: 20px;
    color: #FFF;
    cursor: pointer;
    border: none;
    // position: absolute;
    // bottom: 0;

}