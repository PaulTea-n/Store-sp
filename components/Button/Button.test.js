import Button from "./Button";
import { render, screen, fireEvent } from "@testing-library/react";

const handleClick = jest.fn();

describe("Render Button", () => {
  test("should button render", () => {
    const { asFragment } = render(
      <Button type={"submit"} className="btn">
        Add to cart
      </Button>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Handle click at a button", () => {
  test("should handleClick work", () => {
    render(
      <Button type={"submit"} className="btn" handleClick={handleClick}>
        Add to cart
      </Button>
    );
    const btn = screen.getByText("Add to cart");
    fireEvent.click(btn);
    expect(handleClick).toHaveBeenCalled();
  });
});
