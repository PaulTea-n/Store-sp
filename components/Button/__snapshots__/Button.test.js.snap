// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button should button render 1`] = `
<DocumentFragment>
  <button
    class="btn"
    style="background-color: red;"
    type="submit"
  >
    Add to cart
  </button>
</DocumentFragment>
`;
